pkg/ibm/ibmp.go

```diff
@@ -140,7 +119,52 @@ func (r IBMPowerDynamicConfig) GetInstanceAddress(kubeClient client.Client, ctx
 }
 
 func (r IBMPowerDynamicConfig) ListInstances(kubeClient client.Client, ctx context.Context, instanceTag string) ([]cloud.CloudVMInstance, error) {
-	return nil, fmt.Errorf("not impelemented")
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info("Listing instances by tag:" + instanceTag)
```

Comment:
use structured log values instead of embedding into the string
```suggestion
	log.Info("Listing instances", "tag", instanceTag)
```